### DISCUSSION TITLE
ConfigurationChangeEvent : how to know the change is user or workspace ?

### DISCUSSION THREAD

**Discussion by tksj05** [microsoft/vscode-discussions#124] [Extension Development QnA] [answered]
How to know the change is global or workspace, is there any method ?

I wang to listen the configuration change, i find the ConfigurationChangeEvent, but it seems can not know it is a user configuration change or a workspace configuartion

**Comment by alefragnani** [comment, marked as answer]
You are right, the `ConfigurationChangeEvent` doesn't provide you this level of information.

You can, however, combine with `WorkspaceConfiguration.inspect()` function, to validate each info (`default`, `global`, `workspace`), trying to understand which one was changed.

If this does not work for you, I suggest you to open a feature request in VS Code repo.

Hope this helps
